include/zephyr/bluetooth/audio/csis.h

```diff
@@ -93,6 +96,15 @@ struct bt_csis_cb {
 	 * @return A BT_CSIS_READ_SIRK_REQ_RSP_* response code.
 	 */
 	uint8_t (*sirk_read_req)(struct bt_conn *conn, struct bt_csis *csis);
+
+	/**
+	 * @brief Callback whenever the RSI changes.
+	 *
+	 * If this callback is not set, CSIS will handle advertising of the RSI.
+	 *
+	 * @param prsi Pointer to the new RSI data to be advertised.
+	 */
+	void (*prsi_changed)(const uint8_t *prsi);
```

Comment:
I think @jhedberg suggested that we use `[]` instead of `*` in the API when dealing with arrays. 

```suggestion
	void (*prsi_changed)(const uint8_t prsi[]);
```